src/lcec_el7041_1000.c

```diff
@@ -0,0 +1,713 @@
+    {0x6000, 0x04,  1}, /* Counter underflow */
+    {0x6000, 0x05,  1}, /* Counter overflow */
+    {0x0000, 0x00,  2}, /* Gap */
+    {0x6000, 0x08,  1}, /* Extrapolation stall */
+    {0x6000, 0x09,  1}, /* Status of input A */
+    {0x6000, 0x0a,  1}, /* Status of input B */
+    {0x6000, 0x0b,  1}, /* Status of input C */
+    {0x0000, 0x00,  1}, /* Gap */
+    {0x6000, 0x0d,  1}, /* Status of extern latch */
+    {0x6000, 0x0e,  1}, /* Sync error */
+    {0x0000, 0x00,  1}, /* Gap */
+    {0x6000, 0x10,  1}, /* TxPDO Toggle */
+    {0x6000, 0x11, 16}, /* Counter value */
+    {0x6000, 0x12, 16}  /* Latch value */
+};
+
+static ec_pdo_entry_info_t lcec_el7041_1000_channel1_dcm_in[] = {
+    {0x6010, 0x01,  1}, /* Ready to enable */
+    {0x6010, 0x02,  1}, /* Ready */
+    {0x6010, 0x03,  1}, /* Warning */
+    {0x6010, 0x04,  1}, /* Error */
+    {0x6010, 0x05,  1}, /* Moving positive */
+    {0x6010, 0x06,  1}, /* Moving negative */
+    {0x6010, 0x07,  1}, /* Torque reduced */
+    {0x0000, 0x00,  4}, /* Gap */
+    {0x6010, 0x0c,  1}, /* Digital input 1 */
+    {0x6010, 0x0d,  1}, /* Digital input 2 */
+    {0x6010, 0x0e,  1}, /* Sync error */
+    {0x0000, 0x00,  1}, /* Gap */
+    {0x6010, 0x10,  1}  /* TxPDO Toggle */
+};
+
+static ec_pdo_info_t lcec_el7041_1000_pdos_out[] = {
+    {0x1600, 7, lcec_el7041_1000_channel1_enc_out}, /* ENC RxPDO-Map Control compact */
+    {0x1602, 5, lcec_el7041_1000_channel1_dcm_out}, /* STM RxPDO-Map Control */
+    {0x1604, 1, lcec_el7041_1000_channel1_vel_out}  /* STM RxPDO-Map Velocity */
+};
+
+static ec_pdo_info_t lcec_el7041_1000_pdos_in[] = {
+    {0x1a00, 17, lcec_el7041_1000_channel1_enc_in}, /* ENC TxPDO-Map Status compact */
+    {0x1a03, 13, lcec_el7041_1000_channel1_dcm_in}  /* STM TxPDO-Map Status */
+};
+
+static ec_sync_info_t lcec_el7041_1000_syncs[] = {
+    {0, EC_DIR_OUTPUT, 0, NULL},
+    {1, EC_DIR_INPUT,  0, NULL},
+    {2, EC_DIR_OUTPUT, 3, lcec_el7041_1000_pdos_out},
+    {3, EC_DIR_INPUT,  2, lcec_el7041_1000_pdos_in},
+    {0xff}
+};
+
+void lcec_el7041_1000_read(struct lcec_slave *s, long period);
+void lcec_el7041_1000_write(struct lcec_slave *s, long period);
+
+int lcec_el7041_1000_init(int comp_id, struct lcec_slave *s, ec_pdo_entry_reg_t *r) {
+  lcec_master_t *m = s->master;
+  lcec_el7041_1000_data_t *hal_data;
+  lcec_el7041_1000_chan_t *c;
+  int err;
+
+  // initialize callbacks
+  s->proc_read  = lcec_el7041_1000_read;
+  s->proc_write = lcec_el7041_1000_write;
+
+  // alloc hal memory
+  if ((hal_data = hal_malloc(sizeof(lcec_el7041_1000_data_t))) == NULL) {
+    rtapi_print_msg(RTAPI_MSG_ERR, LCEC_MSG_PFX "hal_malloc() for slave %s.%s failed\n", m->name, s->name);
+    return -EIO;
+  }
+  memset(hal_data, 0, sizeof(lcec_el7041_1000_data_t));
+  s->hal_data = hal_data;
+
+  // initialize sync info
+  s->sync_info = lcec_el7041_1000_syncs;
+
+  // initialize global data
+  hal_data->last_operational = 0;
+
+  // initialize pins
+  c = &hal_data->chans[0];
```

Comment:
This device is single channel. Why to use a channel array? Prepared for future extensions?